### ISSUE TITLE
Short string regressions

### ISSUE THREAD

**Issue by XVilka** [radare/radare2#10581]
Seems both unicode and str.imm commits combined produced regressions in:
```
$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-15476iF83UxKk4C72.tmp /home/travis/build/radare/radare2/radare2-regressions/bins/elf/strenc
e asm.bytes=false
e asm.cmt.off=false
e str.escbslash=true
e asm.strenc=utf16le
pd 1 @ 0x004016de
e asm.strenc=utf32le
pd 1 @ 0x004016de
e asm.filter=false
e asm.noisy=false
e asm.strenc=guess
pd 1 @ 0x004016ed
pd 1 @ 0x004016f7
pd 1 @ 0x00401701
e asm.strenc=utf32le
pd 1 @ 0x00401701
             0x004016de      mov esi, str.Linux_wide__esc:__e_0m        ; u"\t"
             0x004016de      mov esi, str.Linux_wide__esc:__e_0m        ; U"\tLinux_wide\\esc: \x1b[0m\xa1\r\n"
             0x004016ed      mov edi, 0x40258c                          ; U"utf32le> \\u00a2\\u20ac\\U00010348 in cyan:\x1b[36m \xa2\u20ac\U00010348 \x1b[0m\n"
             0x004016f7      mov edi, 0x40266c                          ; U"Mountain range with embedded quad zeros: \U00010300A\U00010300A\U00010300A\n"
-            0x00401701      mov edi, 0x402730                          ; "e%"
-            0x00401701      mov edi, 0x402730                          ; U"\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u256b\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u256b\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u256b\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u256b\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u256b\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u2565\u256b\u2565\u2565\u2565\u2565\u2565\u2565\u2565"
+            0x00401701      mov edi, 0x402730                          ; '0'@' ; "e%"
+            0x00401701      mov edi, 0x402730                          ; '0'@' ; U"\u2565\u2565\u2565\u2565
[XX] db/cmd/cmd_pd asm.strenc utf16le  964
$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -i /tmp/tmp-15476Px6GFv9Tc0pd.tmp /home/travis/build/radare/radare2/radare2-regressions/bins/elf/strenc
e asm.bytes=false
e asm.filter=false
e asm.noisy=false
e str.escbslash=true
e asm.strenc=guess
pd 1 @ 0x004016ac
e asm.filter=true
e asm.cmt.off=false
e asm.strenc=guess
pd 1 @ 0x004016b6
e asm.strenc=utf16le
pd 1 @ 0x004016b6
e asm.strenc=guess
pd 1 @ 0x004016c0
e asm.strenc=utf16le
pd 1 @ 0x004016c0
pd 1 @ 0x004016ca
pd 1 @ 0x004016d4
             0x004016ac      mov edi, 0x40224a                          ; u"utf16le> \\u00a2\\u20ac\\U00010348 in green:\x1b[32m \xa2\u20ac\U00010348 \x1b[0m\n"
             0x004016b6      mov edi, str.e_e_b                         ; "_%e%e%b% "
             0x004016b6      mov edi, str.e_e_b                         ; u"\u255f\u2565\u2565\u2562 is a wall with no embedded zeros\n"
             0x004016c0      mov edi, 0x40230c                          ; " %B%"
             0x004016c0      mov edi, 0x40230c                          ; u"\u2520\u2542\u2500\u2500\u2542\u2528 is a fence with embedded zeros\n"
-            0x004016ca      mov edi, 0x40235a                          ; u"\u2520\u2542-\u2500-\u2500\u2542\u2528 is a fence with embedded double zeros\n"
+            0x004016ca      mov edi, 0x40235a                          ; 'Z#@' ; u"\u2520\u2542-\u2500-\u2500\u2542\u2528 is a fence with embedded double zeros\n"
             0x004016d4      mov edi, 
[XX] db/cmd/cmd_pd r_str_escape anal addr  1214
```
```

**Comment by kazarmy** [comment]
If you set a flag at the address, the short string goes away. I think this behavior should be encouraged so I don't think this needs fixing.

**Comment by kazarmy** [comment]
I've merged in radare/radare2-regressions#1379 so that Travis isn't upset by the above.
